src/iinit.cpp

```diff
@@ -66,11 +69,226 @@ mkrodsdir() {
     return 0; /* no error messages as it normally fails */
 }
 
-void
-printUpdateMsg() {
-    printf( "One or more fields in your iRODS environment file (irods_environment.json) are\n" );
-    printf( "missing; please enter them.\n" );
-}
+namespace
+{
+    auto save_updates_to_irods_environment(const nlohmann::json& _update) -> void
+    {
+        std::string env_file;
+        std::string session_file;
+        if (auto ret = irods::get_json_environment_file(env_file, session_file); !ret.ok()) {
+            fmt::print("failed to get environment file - [{}]\n", ret.code());
+        }
+
+        json obj_to_dump;
+
+        if (std::ifstream in{env_file}; in) {
+            try {
+                in >> obj_to_dump;
+            }
+            catch (const json::parse_error& e) {
+                obj_to_dump = _update;
+                fmt::print(
+                    stderr,
+                    "Failed to parse environment file: []\n"
+                    "Falling back to original environment settings.",
+                    e.what());
+            }
+
+            obj_to_dump.merge_patch(_update);
+        }
+        else {
+            obj_to_dump = _update;
+        }
+
+        if (std::ofstream f(env_file.c_str(), std::ios::out); f.is_open()) {
+            f << obj_to_dump.dump(4) << "\n";
+            f.close();
```

Comment:
This line could be removed.

Is there a reason for explicitly calling `f.close()`?

Reply:
I believe it's because I put the `ofstream` declaration in the if statement and didn't update the logic accordingly. :) Will update.